barretenberg/cpp/src/barretenberg/common/zip_view.hpp

```diff
@@ -122,9 +122,13 @@ class zip_iterator {
     T iter, iter_end;
     std::size_t index = 0;
 
+    // This indirection over std::ref fixes iota_view() and other non-reference containers
+    static auto ref(const auto& t) { return t; }
+    static auto ref(auto& t) { return std::ref(t); }
+    static auto ref(auto&& t) { return std::ref(t); }
```

Comment:
It's nice to zip_view iota view to get [i, elem] type of loops with both index and element, particularly for polynomial where indices don't start from 0